<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Megan Ewing">
<meta name="dcterms.date" content="2024-08-07">

<title>mewing-notebook - August Daily Posts</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">mewing-notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com" rel="" target=""><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">August Daily Posts</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">daily</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Megan Ewing </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">August 7, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#august-7th" id="toc-august-7th" class="nav-link active" data-scroll-target="#august-7th">August 7th</a></li>
  <li><a href="#august-13th" id="toc-august-13th" class="nav-link" data-scroll-target="#august-13th">August 13th</a></li>
  <li><a href="#august-14th" id="toc-august-14th" class="nav-link" data-scroll-target="#august-14th">August 14th</a></li>
  <li><a href="#august-15th" id="toc-august-15th" class="nav-link" data-scroll-target="#august-15th">August 15th</a></li>
  <li><a href="#august-17th" id="toc-august-17th" class="nav-link" data-scroll-target="#august-17th">August 17th</a></li>
  <li><a href="#august-21st" id="toc-august-21st" class="nav-link" data-scroll-target="#august-21st">August 21st</a></li>
  <li><a href="#august-26th" id="toc-august-26th" class="nav-link" data-scroll-target="#august-26th">August 26th</a></li>
  <li><a href="#august-28th" id="toc-august-28th" class="nav-link" data-scroll-target="#august-28th">August 28th</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<section id="august-7th" class="level3">
<h3 class="anchored" data-anchor-id="august-7th">August 7th</h3>
<p>August goals, submitted masters committee milestone. Based off of what Mac and i discussed last week, I went back to trying different pre-filtering parameters and decided to take a different approach. Created histogram, boxplot, and generated summary stats for all of the count data to see how they were spread. We knew the rRNA was eating up a ton of the reads, so I wanted to visually see the spread a bit better, where I could ignore the rRNA interference and make a decision on what made sense as a cutoff for pre-filtering. Depending on if I set the cutoff for the histogram and boxplot counts ranging between 0-50, 0-100, 0-1000, 0-6000, or all counts, the median stayed between 20 to 30. When using all counts the median was 23.</p>
<p><img src="images/0807-histogram_of_read_counts_1to100.png" class="img-fluid" width="340"><img src="images/boxplot_of_all_read_counts.png" class="img-fluid" width="341"></p>
<p>I set the pre-filter parameters to at least 23 counts for at least 8 individuals (so, at least half) for control OR treatment group.</p>
<p><img src="images/0807-DEG_heatmap_8ind23r_ToC.png" class="img-fluid" width="495"></p>
<p>The control and treatment groups stayed together on the heatmap, even when clustered, which is promising.</p>
<p><img src="images/0807-DEG_heatmap_8ind23r_ToC_clust.png" class="img-fluid" width="498"></p>
<p>Waiting on egg quality / mother quality data to make some sense of the samples that still seem to not show super distinct differences in expression.</p>
</section>
<section id="august-13th" class="level3">
<h3 class="anchored" data-anchor-id="august-13th">August 13th</h3>
<p>Small day. Got the egg quality data and started to review it. Bit more info than anticipated. Also organized different degrees of “outliers” (not really outliers) of the samples that didn’t have super distinct expression, based off of the clustering in the heatmap from last week to give an idea what samples to keep an eye out for when diving into the egg data.</p>
</section>
<section id="august-14th" class="level3">
<h3 class="anchored" data-anchor-id="august-14th">August 14th</h3>
<p>Identifying patterns in egg quality data as it relates to the samples in the heatmap. But found that there wasn’t enough egg quality info about the samples that are the “outliers” since not all mothers had egg quality data measured. Results inconclusive.</p>
<p><img src="images/Screen Shot 2024-08-17 at 5.36.54 PM.png" class="img-fluid" width="492"></p>
</section>
<section id="august-15th" class="level3">
<h3 class="anchored" data-anchor-id="august-15th">August 15th</h3>
<p>Switched to looking for patterns in read QC. ie, looking if total number of aligned reads explained any of the data in our heatmap (thinking, “maybe those with overall less reads are the ones that have less distinct expression compared to others in their treatment group”). no dice.</p>
<p><img src="images/Screen Shot 2024-08-17 at 5.43.14 PM.png" class="img-fluid" width="478"></p>
<p>Colors represent the clustering of samples, where more blue shows “cooler” expression on the heatmap for the chunk of genes that are generally “hotter” in the control group. More red bars = “hotter” on the heatmap.</p>
</section>
<section id="august-17th" class="level3">
<h3 class="anchored" data-anchor-id="august-17th">August 17th</h3>
<p>After talking to mac, going to proceed towards annotation with the latest gene list (from aug 7). Clustering looks good, pre filter parameters have sound biological reasoning, and nothing seems to be explaining the differences in expression (or lack of differences) that relates to egg or read quality.</p>
<p>Had an issue previously with blast not having hits for all of my genes in my DEG list, which is odd since the genome is fully annotated. Mac suggested try running with less stringent e value. Went from E-20 to E-5 which feels drastic, but I do filter out for just the hit with the best e value for each LOC, and can use bit score to confirm integrity of the results later downstream.</p>
</section>
<section id="august-21st" class="level3">
<h3 class="anchored" data-anchor-id="august-21st">August 21st</h3>
<p>Got the blast results and took a peak, still not getting all the matches (left hand columns are DEG stats, right hand columns are from the blast output):</p>
<p><a href="https://github.com/meganewing/clamgonads-macsamples/blob/main/code/10-blast_with_cds.Rmd">blast code:</a></p>
<p><img src="images/Screen Shot 2024-08-28 at 10.34.45 AM.png" class="img-fluid" width="394"></p>
<p>output (after <a href="https://github.com/meganewing/clamgonads-macsamples/blob/main/code/12.2-deg_annot_cds_gene_level.Rmd">joining</a>):</p>
<p><img src="images/Screen Shot 2024-08-28 at 10.31.56 AM.png" class="img-fluid"></p>
<p>Not really sure what is going on and if this is expected or not.</p>
<p>Busy week with other jobs, didn’t really have time to do anything besides glance at this and give it one go.</p>
</section>
<section id="august-26th" class="level3">
<h3 class="anchored" data-anchor-id="august-26th">August 26th</h3>
<p>Per Mac’s advice, I went onto NCBI where I could manually search for each of the LOC (aka, “symbol”s) within the <a href="https://www.ncbi.nlm.nih.gov/datasets/gene/GCF_026571515.1/">list of annotated genes for r.phil</a>, and all the ones that had NAs after joining had matches, so I still can’t figure out why there are NAs when going the blast route. Steven says its “to be expected” with species like this, but I still don’t really understand why.</p>
<p>(<a href="https://github.com/RobertsLab/resources/issues/1953">relevant github issue</a>)</p>
<p>(<a href="https://github.com/meganewing/clamgonads-macsamples/blob/main/code/12.2-deg_annot_cds_gene_level.Rmd">relevant code file</a> ; blast code linked above, if needed)</p>
<p>I revised the code to join the DEG data to a tsv of the <a href="https://www.ncbi.nlm.nih.gov/datasets/gene/GCF_026571515.1/">annotated genes from NCBI</a>, which paired data to all of my LOC (no NA’s), but there’s no SPID (uniprot accession numbers) associated with the data.</p>
<p>I want to be able to move onto gene ontology, but not sure how to do so without uniprot accession numbers.</p>
<p>To resolve this I tried two approaches:</p>
<p><strong>FIRST:</strong></p>
<p>While no SPID, there was however, protein accession numbers. Since this genome annotation was RefSeq, I changed the ‘from database’ and used the protein accession numbers. To no avail. It mapped to UniParc, which provided no GO information</p>
<p><img src="images/Screen Shot 2024-08-28 at 10.43.24 AM.png" class="img-fluid" width="270"></p>
<p><img src="images/Screen Shot 2024-08-28 at 10.42.31 AM.png" class="img-fluid" width="232"></p>
<p><strong>SECOND</strong>:</p>
<p>I tried re joining my new DEG/annotation joined list to the full blast results by protein ID/accession number. This still left me with NAs, even for characterized proteins.</p>
<p><img src="images/Screen Shot 2024-08-28 at 10.48.02 AM.png" class="img-fluid"></p>
<p>So I’m not sure where to go from here.</p>
<p><strong>I want to be able to get GO information so I can get biological process information.</strong></p>
</section>
<section id="august-28th" class="level3">
<h3 class="anchored" data-anchor-id="august-28th">August 28th</h3>
<ul>
<li>Provide preview (<code>head</code>) of BLAST output table.</li>
</ul>
<p><img src="images/Screen Shot 2024-08-28 at 2.01.37 PM.png" class="img-fluid"></p>
<ul>
<li>Provide preview of your query file(s) for BLASTing.</li>
</ul>
<p><em>I don’t have this right now because it’s too big to push and my computer doesn’t work with the VPN. Will update when I can go to the office tomorrow. It is on raven though and the relative path is</em> <code>../data/ncbi_dataset/data/GCF_026571515.1/cds_from_genomic.fna</code></p>
<ul>
<li>Provide preview (<code>head</code>) of <code>0807-DEGstats_ToC_8ind23r.tab</code>.</li>
</ul>
<p><img src="images/Screen Shot 2024-08-28 at 2.05.56 PM.png" class="img-fluid"></p>
<ul>
<li>Provide preview (<code>head</code>) of <code>ncbi_dataset.tsv</code>.</li>
</ul>
<p><img src="images/Screen Shot 2024-08-28 at 2.12.55 PM.png" class="img-fluid"></p>
<p>col 1-9</p>
<p><img src="images/Screen Shot 2024-08-28 at 2.13.17 PM.png" class="img-fluid"></p>
<p>col 9-15</p>
<p>Including step-wise previews of the various tables/files will greatly help.</p>
<p><a href="https://rpubs.com/mewing0/1214571" class="uri">https://rpubs.com/mewing0/1214571</a></p>
<p>Also, a clear statement on what data you’re starting with and what you want for an end result will help.</p>
<p><em>the workflow, generally speaking, is as follows (starting from featurecounts)</em></p>
<ol type="1">
<li><p><em>featurecounts -&gt;</em></p></li>
<li><p><em>filter out any genes that don’t meet the parameters (see aug. 7) -&gt;</em></p></li>
<li><p><em>DESeq2 -&gt; filter for significant padj -&gt;</em></p></li>
<li><p><em>generated a dataframe that has the DEGs and corresponding stats (see above) &amp; a dataframe that has DEGs and corresponding counts for each sample (not super relevant) -&gt;</em></p></li>
<li><p><em><strong>WANT:</strong> to connect annotation information to DEG list that will allow me to get GO terms (whether through uniprot accession numbers/SPID or other methods unknown to me)</em></p></li>
</ol>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>